crates/bevy_asset/src/handle.rs

```diff
@@ -74,6 +74,13 @@ impl HandleId {
     }
 }
 
+/// A type for handler-related errors.
+#[derive(Debug)]
+pub enum HandleError {
```

Comment:
I'd like to use the `thiserror` crate to get nice error messages and implement the `Error` trait on our error types. We use it elsewhere, and should do so here.